config_db.go

```diff
@@ -136,95 +136,169 @@ func byIDs(ids []string) bson.M {
 }
 
 // getSectionsBSON returns the config documents from the database as a slice of [bson.Raw].
-// Configuration is always fetched from the shared database. If includeOverrides is true
-// the config documents are first fetched from the local database and only fetched from
-// the shared database when they're missing from the local database. This means that a
-// local documents will override the same document from the shared database.
+// If no shared database exists all configuration is fetched from the local database. If a
+// shared database exists, configuration is fetched from the shared database, save for the
+// overrides config document which is fetched from the local database and, if [includeOverrides],
+// the overrides in the overrides config document are applied to the rest of the configuration.
 func getSectionsBSON(ctx context.Context, ids []string, includeOverrides bool) ([]bson.Raw, error) {
-	missingIDs := ids
+	db := GetEnvironment().DB()
+	if sharedDB := GetEnvironment().SharedDB(); sharedDB != nil {
+		db = sharedDB
+	}
+
+	cur, err := db.Collection(ConfigCollection).Find(ctx, byIDs(ids))
+	if err != nil {
+		return nil, errors.Wrap(err, "finding local configuration sections")
+	}
 	docs := make([]bson.Raw, 0, len(ids))
-	if includeOverrides {
-		cur, err := GetEnvironment().DB().Collection(ConfigCollection).Find(ctx, byIDs(ids))
+	if err := cur.All(ctx, &docs); err != nil {
+		return nil, errors.Wrap(err, "iterating cursor for local configuration sections")
+	}
+
+	if GetEnvironment().SharedDB() != nil {
+		docs, err = overrideConfig(ctx, docs, includeOverrides)
 		if err != nil {
-			return nil, errors.Wrap(err, "finding local configuration sections")
+			return nil, errors.Wrap(err, "overriding config")
 		}
+	}
+
+	return docs, nil
+}
 
-		if err := cur.All(ctx, &docs); err != nil {
-			return nil, errors.Wrap(err, "iterating cursor for local configuration sections")
+// overrideConfig overrides the content of the overrides document with the contents of the overrides
```

Comment:
This comment is confusing, I'm not sure I get the point of the function (maybe too many overrides). Could you update it?

Reply:
Updated the comment and rejiggered the code in an attempt to demystify it.